Assets/Scripts/ScrapCollector.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ScrapCollector : MonoBehaviour
+{
+    [SerializeField] private ScrapManager scrapManager;
+    [SerializeField] private List<PlayerController> enemyMecha;
+    [SerializeField] private List<PlayerController> highHealthMecha;
+    private int scrapAvailable;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        CollectEnemyMecha();
+    }
+
+    private void Update()
+    {
+        scrapAvailable = scrapManager.GetScrapAvailable();
+    }
+
+    public void CollectEnemyMecha()
+    {
+        GameObject[] boardEntities = GameObject.FindGameObjectsWithTag("Player");
+
+        foreach (GameObject entity in boardEntities)
+        {
+            PlayerController playerController = entity.GetComponent<PlayerController>();
+            if (playerController.isPlayerEntity == false)
```

Comment:
This assumption is probably fine, but it could be incorrect if there are non-player controlled allies.

Reply:
True. I'll check with Remy to see if there could be a case in which that would happen.

Reply:
I spoke with Remy, and he said that we haven't implemented AI that would include non-player controlled allies yet. I'll leave a comment about that for anyone who might need to include that later on -- how does that sound?